service-base/src/test/java/org/eclipse/hono/service/tenant/BaseTenantServiceTest.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.tenant;
+
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static org.eclipse.hono.util.TenantConstants.Action;
+import static org.mockito.Mockito.*;
+
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.util.*;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+
+/**
+ * Tests verifying behavior of {@link BaseTenantService}.
+ *
+ */
+@RunWith(VertxUnitRunner.class)
+public class BaseTenantServiceTest {
+    private static final String TEST_TENANT = "dummy";
+    private static BaseTenantService<ServiceConfigProperties> tenantService;
+
+    /**
+     * Sets up the fixture.
+     */
+    @BeforeClass
+    public static void setUp() {
+        tenantService = createBaseTenantService();
+    }
+
+    /**
+     * Verifies that the base service accepts a request for adding
+     * a tenant that contains the minimum required properties.
+     */
+    @Test
+    public void testAddSucceedsForMinimalData() {
+
+        final JsonObject testPayload = createValidTenantPayload();
+
+        final Message<JsonObject> msg = createMessageMockForPayload(Action.ACTION_ADD, testPayload);
+        tenantService.processTenantMessage(msg);
+
+        verify(msg).reply(resultWithStatusCode(HTTP_CREATED, TEST_TENANT));
+    }
+
+    /**
+     * Verifies that the base service accepts a request for adding
```

Comment:
this description does not match the test case impementation

Reply:
:+1: